packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -2842,10 +2864,7 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin {
             () => <_RouteEntry>[]
         );
         pagelessRoutes.add(potentialEntryToRemove);
-        assert(() {
-          potentialEntryToRemove._debugWaitingForExitDecision = previousOldPageRouteEntry._debugWaitingForExitDecision;
-          return true;
-        }());
+        potentialEntryToRemove._isWaitingForExitingDecision = previousOldPageRouteEntry.isWaitingForExitingDecision;
```

Comment:
as per comment above, this would then be:

potentialEntryToRemove.markNeedsExitingDecision();